### PR TITLE
[Network Drive] Fix unknown enum error: 0xc0000070

### PR DESCRIPTION
**Pull request by akanshi-elastic**
## Closes https://github.com/elastic/connectors/issues/2356###

This PR includes fix for getting SMBResponseException: Received unexpected status from the server: Unknown error. ([3221225584](tel:3221225584)) UNKNOWN_ENUM: 0xc0000070.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

**Comment by akanshi-elastic** [comment]
Buildkite test this

File: connectors/sources/network_drive.py [hunk 1]
```diff
@@ -62,6 +73,26 @@
...
+class InvalidLogonHoursException(Exception):
+    pass
+
+
+class PasswordChangeRequiredException(Exception):
+    pass
+
+
+class NoLogonServerException(Exception):
+    pass
```
Comment (artem-shelkovnikov) [inline connectors/sources/network_drive.py:93]: Can we add user-friendly error messages here so that the user can act them?

Comment (akanshi-elastic) [reply connectors/sources/network_drive.py:93]: Error message are already set as per the error on line: https://github.com/elastic/connectors/pull/2458/files#diff-29e12700a1a4cf24af22d5bf0a870c9ebfede0ec41dc991406b951d0100393ffR286

Comment (artem-shelkovnikov) [reply connectors/sources/network_drive.py:93]: Ah okay, misread it.

**Review by artem-shelkovnikov — COMMENTED** [review]
To make sure - does all of these errors happen during connection, or some can happen during access to specific resource?

**Comment by akanshi-elastic** [comment]
> To make sure - does all of these errors happen during connection, or some can happen during access to specific resource?

yes, all of them occur during connection only.

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**